### PR TITLE
Outbound traffic in Overview cards. Added dropdown.

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/4896

Added Traffic Direction dropdown in Overview toolbar to select Outbound and Inbound traffic to show. Selection triggers reload of Overview items and the traffic with particular direction.

Added cypress tests.

Screenshot from 2022-05-25 18-04-08

### CODE REVIEW

File: frontend/cypress/integration/common/overview.ts [hunk 1]
```diff
@@ -125,6 +125,26 @@ When(`user selects {string} time range`, (interval) => {
         .should('not.exist');
 });
 
+When(`user selects {string} traffic duration`, (duration) => {
```
Comment (leandroberetta) [inline frontend/cypress/integration/common/overview.ts:128]: Typo, I guess it should be direction.

I guess we can add the type to the parameter? (direction: string)

**Review by leandroberetta — COMMENTED** [review]
@hhovsepy very nice addition to the overview page. 

Everything LGTM, only some minor comments, one is below and the other is an aesthetic comment:

overview_in_out_traffic

I guess we need more space between the chart and the "ops" axis label, but also following Andrew's comment on the issue, more space separating the section above.

**Comment by lucasponce** [comment]
Minor:
image

"Traffic direction" -> "Traffic"

I think that in the chart we use the "inbound/outbound traffic" then the text can be shorter, which would help to not have a very large populated toolbar

**Comment by lucasponce** [comment]
Some note, I'm reviewing the graph vs the overview to check that information is synced:
image

Queries used are correct:
image

But I saw some gap in the Graph sidebar, "total" is populated:
image


But, "inbound" and "outbound" are empty:
image
image

It sounds a regression for @jshaughn but it's out of the scope of this PR, just leaving the comment here.

**Review by lucasponce — CHANGES_REQUESTED** [review]
LGTM
A minor suggestion to "shorten" the name to avoid a populated toolbar, but the work looks good.

**Comment by hhovsepy** [comment]
Have updated the paddings and the toolbar text.

**Comment by hhovsepy** [comment]
"MeshConfig.OutboundTrafficPolicy.Mode" flag will be added in a scope of different PR.

**Comment by hhovsepy** [comment]
Screenshot from 2022-05-27 12-45-37

**Review by lucasponce — APPROVED** [review]

**Comment by jshaughn** [comment]
@lucasponce , I'll look at that graph side panel.  One comment for you and @hhovsepy re: this PR:  Perhaps the new dropdown should be disabled when using small cards, because they have no graph?

**Comment by jshaughn** [comment]
> It sounds a regression for @jshaughn but it's out of the scope of this PR, just leaving the comment here.

Actually this is working as expected.  There is no `Incoming` traffic into the selected namespace, and no outgoing traffic from the namespace.  The total reflects the traffic within the namespace.

**Comment by lucasponce** [comment]
Then this picture:
image

The "total" is the sum of the inbound of the workload and the outbound of the workloads, but it doesn't necessarily mean out of the namespace.

am I right?

**Comment by jshaughn** [comment]
I believe the numbers and sparklines are all as intended.  The reason it's confusing is that "Inbound" and "Outbound" have some context-specific semantics, and there is currently no tooltip or other help to actually explain what it means in the different places.  

For example, for the full graph side-panel, "Inbound" is defined as traffic originating from a traffic source, i.e. the traffic flowing into the graph.  This makes sense.   For the namespace side-panel "Inbound" means traffic entering the namespace from outside the namesapce.  This also makes sense.  "Total" in all cases (I think) means all of the traffic for the object under observation.

Note that the community has never really complained, so I'm not sure they care, or look carefully at some of this stuff.  And I think we should prioritize changes accordingly (meaning I'm not in favor of a lot of work here).

**Comment by lucasponce** [comment]
> Actually this is working as expected. There is no Incoming traffic into the selected namespace and no Outgoing traffic from the namespace. The total reflects the traffic within the namespace.

I was thinking more of a minor tooltip to explain this, here and potentially on the overview page. The queries look right but I think that contextual information should deserve some notes.

**Pull request merged**